CRM/SMS/Provider.php

```diff
@@ -207,7 +207,7 @@ public function processInbound($from, $body, $to = NULL, $trackID = NULL) {
     if (!$message->toContactID) {
       // find recipient if $toContactID not set by hook
       if ($message->to) {
-        $message->toContactID = CRM_Core_DAO::singleValueQuery("SELECT contact_id FROM civicrm_phone JOIN civicrm_contact ON civicrm_contact.id = civicrm_phone.contact_id WHERE !civicrm_contact.is_deleted AND phone LIKE %1", [
+        $message->toContactID = CRM_Core_DAO::singleValueQuery("SELECT contact_id FROM civicrm_phone JOIN civicrm_contact ON civicrm_contact.id = civicrm_phone.contact_id WHERE !civicrm_contact.is_deleted AND phone_numeric LIKE %1", [
           1 => ['%' . $message->to, 'String'],
```

Comment:
Off the top of my head, this is now comparing message->to to phone_numeric, but that's unlikely to match.